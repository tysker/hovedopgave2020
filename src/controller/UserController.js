import React from 'react';
import styled from "styled-components";

const UserController = ({addStraight, addCurve, rotateTrack, deleteLastTrack}) => {
    return (
        <div>
            <ButtonGrid>
                <button data-testid='add-straight' className="btn" onClick={addStraight}>ADD TRACK</button>
                <button data-testid='add-curve' className="btn" onClick={addCurve}>ADD CURVE</button>
                <button data-testid='rotate-track' className="btn" onClick={rotateTrack}>ROTATE</button>
                <button data-testid='delete-track' className="btn" onClick={deleteLastTrack}>DELETE</button>
            </ButtonGrid>
        </div>
    );
};

export default UserController;

// Style
const
    ButtonGrid = styled.div`
    display: grid;
    padding: 1rem;
    grid-template-columns: repeat(4, 1fr);
    grid-row-gap: 1rem;
    justify-items: center;
`;