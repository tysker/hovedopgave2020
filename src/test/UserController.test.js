import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import UserController from "../controller/UserController";

afterEach(cleanup);

test('<GraphicController-Buttons/>', () => {
    const {getByTestId, queryByTestId, getByText, debug} = render(<UserController/>);

    //debug();
    console.error = jest.fn();

    const trackButton = getByTestId('add-straight');
    const curveButton = getByTestId('add-curve');
    const addDeleteButton = getByTestId('delete-track');
    const rotateButton = getByTestId('rotate-track');
    const canvasButton = getByTestId('reset-canvas');

    // Test if Test ID's are existing
    expect(queryByTestId('add-straight')).toBeTruthy();
    expect(queryByTestId('add-curve')).toBeTruthy();
    expect(queryByTestId('delete-track')).toBeTruthy();
    expect(queryByTestId('rotate-track')).toBeTruthy();
    expect(queryByTestId('reset-canvas')).toBeTruthy();

    // Asserts track-button is a button
    expect(trackButton.tagName).toBe('BUTTON');
    expect(addDeleteButton.tagName).toBe('BUTTON');
    expect(curveButton.tagName).toBe('BUTTON');
    expect(rotateButton.tagName).toBe('BUTTON');
    expect(canvasButton.tagName).toBe('BUTTON');
    // Asserts track-buttons text content
    expect(trackButton.textContent).toBe('ADD TRACK');
    expect(curveButton.textContent).toBe('ADD CURVE');
    expect(addDeleteButton.textContent).toBe('DELETE');
    expect(rotateButton.textContent).toBe('ROTATE');
    expect(canvasButton.textContent).toBe('RESET');

});

// Snapshot test: creates a snapshot of the DOM
test('<GraphicController-Snapshot/>', () => {
    const {container} = render(<UserController/>);
    expect(container.firstChild).toMatchSnapshot();
});

// Test if the button event triggers the right function
test('<GraphicController-FireEvent/>', () => {
    const track = jest.fn();
    const {getByText} = render(<UserController
        addStraight={track}
        addCurve={track}
        deleteLastTrack={track}
        resetCanvas={track}
        rotateTrack={track}/>);

    fireEvent.click(getByText('ADD TRACK'));
    fireEvent.click(getByText('ADD CURVE'));
    fireEvent.click(getByText('DELETE'));
    fireEvent.click(getByText('ROTATE'));
    fireEvent.click(getByText('RESET'));
    expect(track).toHaveBeenCalledTimes(5);
})


