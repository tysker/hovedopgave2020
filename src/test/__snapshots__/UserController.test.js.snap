// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GraphicController-Snapshot/> 1`] = `
<div>
  <div
    class="sc-AxjAm ensnQq"
  >
    <button
      class="btn"
      data-testid="add-straight"
    >
      ADD TRACK
    </button>
    <button
      class="btn"
      data-testid="add-drawCurve"
    >
      ADD CURVE
    </button>
    <button
      class="btn"
      data-testid="rotate-track"
    >
      ROTATE
    </button>
    <button
      class="btn"
      data-testid="delete-track"
    >
      DELETE
    </button>
    <button
      class="btn"
      data-testid="reset-canvas"
    >
      RESET
    </button>
  </div>
</div>
`;
